### ISSUE TITLE
Ambiguous module reference: kotlinx.coroutines.core

### ISSUE THREAD

**Issue by oryan-block** [Kotlin/kotlinx.coroutines#3842]
**Describe the bug**

I'm trying to upgrade to v1.7.3 and add the modules to a module-info file but it seems that `kotlinx-coroutines-core` and `kotlinx-coroutines-core-jvm` cause a problem because they have the same module name (I think)

![image](https://github.com/Kotlin/kotlinx.coroutines/assets/73128417/c99a88c9-0ec5-4817-b864-c42d96d13d61)

My pom dependencies:

```xml
        <dependency>
            <groupId>org.jetbrains.kotlinx</groupId>
            <artifactId>kotlinx-coroutines-core</artifactId>
        </dependency>
        <dependency>
            <groupId>org.jetbrains.kotlinx</groupId>
            <artifactId>kotlinx-coroutines-jdk9</artifactId>
        </dependency>
```
This seems to happen because `kotlinx-coroutines-core-jvm` is a dependency of `kotlinx-coroutines-jdk9` and it's module name is `kotlinx-coroutines-core`. At the same time `kotlinx-coroutines-core` has an implied module name which is its package name. 
Am I importing these dependencies wrong?

**Comment by oryan-block** [comment]
btw if I replace these deps with `kotlinx-coroutines-core-jvm` the warning is gone, but that doesn't seem to be the way to import the project according to the docs

**Comment by mjovanc** [comment]
@oryan-block Is this still an issue for you? The way I see it you should not add the extension `-jvm`, it should only be `kotlinx-coroutines-core`. The packages for the specific platform will be present and common ones.

I could not find the branch now to test.

**Comment by oryan-block** [comment]
@mjovanc yes this is still a problem. This happens even if I only import `kotlinx-coroutines-core`
Please check this branch: https://github.com/graphql-java-kickstart/graphql-java-tools/tree/coroutinies-module-reference

**Comment by dkhalanskyjb** [comment]
In Maven projects, it's ok to depend on the `-jvm` variant of the Kotlin Multiplatform libraries. In fact, it's more straightforward: all `kotlinx-coroutines-core` should do in Maven is specify a dependency on `kotlinx-coroutines-core-jvm` for convenience. I'm not sure why also having `kotlinx-coroutines-core` in the dependencies causes a warning. Is it only in the IDE, or can you observe it when building the project?

**Comment by oryan-block** [comment]
@dkhalanskyjb it's not just the IDE, maven also fail to build because it can't see anything in the ambiguous module. Depending on `-jvm` fixes it. Should the readme be fixed?

> I'm not sure why also having kotlinx-coroutines-core in the dependencies causes a warning

Not 100% sure bu I do believe it's because both `core` and `core-jvm` end up with the same module name.

**Comment by fzhinkin** [comment]
The issue arise because `kotlinx-coroutines-core.jar` does not contain a JPMS module, so the JPMS module name is derived from the jar file name (`kotlinx.coroutines.core`). And that module name conflicts with the actual `kotlinx.coroutines.core` module when the `kotlinx-coroutines-core-jvm` dependency is fetched.

As a workaround, we can set some automatic module name explicitly using the `Automatic-Module-Name` manifest header.
Something like `Automatic-Module-Name: kotlinx.coroutines.core.trampoline` :)

**Comment by oryan-block** [comment]
@fzhinkin yep, just need to resolve that conflict.
